suite-common/wallet-core/src/accounts/accountsReducer.ts

```diff
@@ -397,9 +401,23 @@ export const selectIsEmptyDevice = (
 
 export const selectHasOnlyEmptyPortfolioTracker = memoize(
     (state: AccountsRootState & DeviceRootState & DiscoveryRootState) => {
-        const isEmptyDevice = selectIsEmptyDevice(state);
+        const isDiscoveredAccountlessDevice = selectIsDiscoveredAccountlessDevice(state);
         const hasOnlyPortfolioDevice = selectHasOnlyPortfolioDevice(state);
 
-        return isEmptyDevice && hasOnlyPortfolioDevice;
+        return isDiscoveredAccountlessDevice && hasOnlyPortfolioDevice;
+    },
+);
+
+// If we know that device is not empty, we can return true right away.
+// Otherwise we return null and wait until discovery finishes
+export const selectDeviceIsNotEmpty = memoize(
```

Comment:
It returns `boolean` or `null`, right? I see no point of memoizing this. And I would change the name:
```suggestion
export const selectIsDeviceNotEmpty = (
```

nitpick: I find `if` conditions without brackets unnecessary harder to read, especially if the condition is longer and it's not just `if(isLoading) return null;` or something like that... But feel free to disagree :-)

Reply:
I actually do feels the same and dont know, why I havent used those. I'll update and remove memoization

Reply:
e6791b5ea8